Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Aos_observer.h

```diff
@@ -0,0 +1,409 @@
+   * now consists only of a the unbounded face `uf`.
+   */
+  virtual void after_clear(Face_handle uf);
+
+  /*! issued just before a global function starts to modify the attached
+   * arrangement. It is guaranteed that no queries (especially no point-location
+   * queries) are issued until the termination of the global function is indicated
+   * by `after_global_change()`.
+   */
+  virtual void before_global_change();
+
+  /*!
+    issued immediately after a global function has stopped modifying the
+    attached arrangement.
+  */
+  virtual void after_global_change();
+
+  /// @}
+
+  /// \name Notifications on Attachment or Detachment
+  /// @{
+
+  /*! issued just before the observer is attached to the arrangement instance
+   * `arr`.
+   */
+  virtual void before_attach(const Arrangement_2& arr);
+
+  /*! issued immediately after the observer has been attached to an arrangement
+   * instance.
+   */
+  virtual void after_attach();
+
+  /*! issued just before the observer is detached from its arrangement instance.
+   */
+  virtual void before_detach();
+
+  /*! issued immediately after the observer has been detached from its arrangement
+   * instance.
+   */
+  virtual void after_detach();
+
+  /// @}
+
+  /// \name Notifications on Local Changes in the Arrangement
+  /// @{
+
+  /*! issued just before a new vertex that corresponds to the point `p` is
+   * created.
+   */
+  virtual void before_create_vertex(const Point_2& p);
+
+  /*! issued immediately after a new vertex `v` has been created.  Note that the
+   * vertex still has no incident edges and is not connected to any other vertex.
+   */
+  virtual void after_create_vertex(Vertex_handle v);
+
+  /*! issued just before a new vertex at infinity is created, `cv` is the curve
+   * incident to the surface boundary, `ind` is the relevant curve-end, `ps_x` is
+   * the boundary condition of the vertex in \f$ x\f$ and `ps_y` is the boundary
+   * condition of the vertex in \f$ y\f$.
+   */
+  virtual void before_create_boundary_vertex(const X_monotone_curve_2& cv,
+                                             Arr_curve_end ind,
+                                             Arr_parameter_space ps_x,
+                                             Arr_parameter_space ps_y);
+
+  /*! issued immediately after a new vertex `v` has been created.  Note that the
+   * vertex still has no incident edges and is not connected to any other vertex.
+   */
+  virtual void after_create_boundary_vertex(Vertex_handle v);
+
+  /*! issued just before a new edge that corresponds to the \f$ x\f$-monotone
+   * curve `c` and connects the vertices `v1` and `v2` is created.
+   */
+  virtual void before_create_edge(const X_monotone_curve_2& c,
+                                  Vertex_handle v1, Vertex_handle v2);
+
+  /*! issued immediately after a new edge `e` has been created.  The halfedge that
+   * is sent to this function is always directed from `v1` to `v2` (see above).
```

Comment:
Better refer to the function name as e.g. due to sorting the order in the documentation can change.